go-controller/pkg/allocator/udn/ips.go

```diff
@@ -0,0 +1,76 @@
+package udn
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	utilnet "k8s.io/utils/net"
+)
+
+const (
+	masqueradeIPv4IDName = "v4-masquerade-ips"
+	masqueradeIPv6IDName = "v6-masquerade-ips"
+)
+
+type MasqueradeIPs struct {
+	Shared net.IP
+	Local  net.IP
+}
+
+// AllocateV4MasqueradeIPs will return local and shared gateway masquerade IPv4 addresses calculated from
```

Comment:
change func description for these as well, I wouldn't tie it to the gateways specifically because both modes somtimes operate interchangeably meaning the gatewayrouter masquerade will be used in lgw for services flows